### PR TITLE
feat: add SIGN, REPLACE and INITCAP

### PR DESCRIPTION
**Pull request by jzaralim**
### Description 
Added the `SIGN`, `REPLACE` and `INITCAP` functions.

### Testing done 
Wrote unit tests and QTT tests for each.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/syntax-reference.rst [hunk 1]
```diff
@@ -1645,8 +1648,16 @@ Scalar functions
...
 +------------------------+---------------------------------------------------------------------------+---------------------------------------------------+
+| REPLACE                |  ``REPLACE(col1, 'foo', 'bar')``                                          | Replace all instances of a substring in a string  |
+|                        |                                                                           | with a new string.                                |
++------------------------+---------------------------------------------------------------------------+---------------------------------------------------+
 | ROUND                  |  ``ROUND(col1)``                                                          | Round a value to the nearest BIGINT value.        |
 +------------------------+---------------------------------------------------------------------------+---------------------------------------------------+
+| SIGN                   |  ``SIGN(col1)``                                                           | Returns the sign of the argument:                 |
```
Comment (blueedgenick) [inline docs/developer-guide/syntax-reference.rst]: should specify not only the possible values here (-1/0/1/null) but also the datatype (should be Int, not Double, see later comment)

File: ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Sign.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+@SuppressWarnings("WeakerAccess") // Invoked via reflection
+@UdfDescription(
+    name = "sign",
+    author = KsqlConstants.CONFLUENT_AUTHOR,
+    description = "The sign of a value."
+)
+public class Sign {
+  @Udf(description = "Returns the sign of an INT value.")
+  public Double sign(
```
Comment (blueedgenick) [inline ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Sign.java]: seems very strange to return Doubles from these when the return value can,  by definition, only be integral ? this kind of counter-intuitive behavior will trip up users

File: docs/developer-guide/syntax-reference.rst [hunk 3]
```diff
@@ -1645,8 +1648,16 @@ Scalar functions
...
+|                        |                                                                           | * -1 if the argument is negative                  |
```
Comment (blueedgenick) [inline docs/developer-guide/syntax-reference.rst]: maybe specify also that it works on _numeric_ input arguments (only)

File: ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Sign.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import io.confluent.ksql.util.KsqlConstants;
+
+@SuppressWarnings("WeakerAccess") // Invoked via reflection
+@UdfDescription(
+    name = "sign",
+    author = KsqlConstants.CONFLUENT_AUTHOR,
+    description = "The sign of a value."
+)
+public class Sign {
+  @Udf(description = "Returns the sign of an INT value.")
```
Comment (blueedgenick) [inline ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Sign.java]: this is the description the user actually sees inside of KSQL when they describe the function so it's useful to repeat here the fact that it returns (-1|0|1) - just saying "it returns the sign" is probably not specific enough

File: ksql-engine/src/main/java/io/confluent/ksql/function/udf/string/InitCap.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import io.confluent.ksql.function.udf.UdfParameter;
+import io.confluent.ksql.util.KsqlConstants;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@SuppressWarnings("unused") // Invoked via reflection.
+@UdfDescription(name = "initcap",
+    author = KsqlConstants.CONFLUENT_AUTHOR,
+    description = "Capitalizes the first letter of each word in a string and the rest lowercased.")
```
Comment (blueedgenick) [inline ksql-engine/src/main/java/io/confluent/ksql/function/udf/string/InitCap.java]: a few words here, and in the syntax reference, about what constitutes a new "word" in the string would be appropriate - it's not as straightforward as one might hope with various unicode strings!

File: docs/developer-guide/syntax-reference.rst [hunk 6]
```diff
@@ -1645,8 +1649,17 @@ Scalar functions
...
 +------------------------+---------------------------------------------------------------------------+---------------------------------------------------+
+| REPLACE                |  ``REPLACE(col1, 'foo', 'bar')``                                          | Replace all instances of a substring in a string  |
+|                        |                                                                           | with a new string.                                |
++------------------------+---------------------------------------------------------------------------+---------------------------------------------------+
 | ROUND                  |  ``ROUND(col1)``                                                          | Round a value to the nearest BIGINT value.        |
 +------------------------+---------------------------------------------------------------------------+---------------------------------------------------+
+| SIGN                   |  ``SIGN(col1)``                                                           | Returns the sign of a numeric value as an INTEGER:|
```
Comment (JimGalasyn) [inline docs/developer-guide/syntax-reference.rst]: "Return the sign of", or maybe just "The sign of".

**Comment by apurvam** [comment]
ping @confluentinc/ksql for reviews!

**Pull request merged**